### ISSUE TITLE
Matrix Multiplication for large SizedArray

### ISSUE THREAD

**Issue by bjack205** [JuliaArrays/StaticArrays.jl#734]
I get the following error when multiplying two SizedArrays
`conversion to pointer not defined for SizedArray{Tuple{20,20},Float64,2,2}`

This occurs when the the `_mul` method passes the arrays to BLAS, and the error happens on the `ccal` in `mul_blas!`.

**Comment by c42f** [comment]
I haven't been able to reproduce this yet. The following works for me:

```julia
julia> a = rand(SizedMatrix{20,20});

julia> a*a
20×20 SArray{Tuple{20,20},Float64,2,400} with indices SOneTo(20)×SOneTo(20):
...
```

(You'll notice it produces an `SArray`, however that is changed in the latest master branch.)

Could you provide a complete self contained example of the failure?

**Comment by bjack205** [comment]
Sorry, my bad, I forgot to mention I'm using the in-place matrix multiply
```julia
julia> a = rand(SizedMatrix{20,20});
julia> b = rand(SizedMatrix{20,20});
julia> c = zeros(SizedMatrix{20,20});
julia> mul!(c,a,b)
ERROR: conversion to pointer not defined for SizedArray{Tuple{20,20},Float64,2,2}
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] unsafe_convert(::Type{Ptr{Float64}}, ::SizedArray{Tuple{20,20},Float64,2,2}) at ./pointer.jl:67
 [3] macro expansion at /home/brian/.julia/packages/StaticArrays/1g9bq/src/matrix_multiply.jl:307 [inlined]
 [4] mul_blas!(::Size{(20, 20)}, ::SizedArray{Tuple{20,20},Float64,2,2}, ::Size{(20, 20)}, ::Size{(20, 20)}, ::SizedArray{Tuple{20,20},Float64,2,2}, ::SizedArray{Tuple{20,20},Float64,2,2}) at /home/brian/.julia/packages/StaticArrays/1g9bq/src/matrix_multiply.jl:289
 [5] macro expansion at /home/brian/.julia/packages/StaticArrays/1g9bq/src/matrix_multiply.jl:266 [inlined]
 [6] _mul! at /home/brian/.julia/packages/StaticArrays/1g9bq/src/matrix_multiply.jl:248 [inlined]
 [7] mul!(::SizedArray{Tuple{20,20},Float64,2,2}, ::SizedArray{Tuple{20,20},Float64,2,2}, ::SizedArray{Tuple{20,20},Float64,2,2}) at /home/brian/.julia/packages/StaticArrays/1g9bq/src/matrix_multiply.jl:17
 [8] top-level scope at REPL[8]:1

```

**Comment by c42f** [comment]
Indeed that's a bug, thanks for the report!

**Comment by c42f** [comment]
Fixed in #738
